plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -217,6 +223,8 @@ func (monitor *DirectoryMonitor) read(filePath string) {
 	// File is finished, move it to the 'finished' directory.
 	monitor.moveFile(filePath, monitor.FinishedDirectory)
 	monitor.filesProcessed.Incr(1)
+	// Backward compatibility
+	selfstat.Register("directory_monitor", "files_processed", nil).Incr(1)
```

Comment:
I don't think we should achieve backward compatibility like this. See my comments below.